File: 283-move-zeroes/move-zeroes.java
class Solution 
{
    public void moveZeroes(int[] nums) 
    {
        if(nums.length <= 1)
        {
            return;
        }
        
        int index = 0;

        if(nums.length > 1)
        {
            for(int num : nums)
            {
                if(num != 0)
                {
                    nums[index++] = num;
                }
            }         
        }

        for(int i = index; i<nums.length; i++)
        {
            nums[i] = 0;
        }
    }
}

File: 283-move-zeroes/move-zeroes.cs
public class Solution 
{
    public void MoveZeroes(int[] nums) 
    {
        int len = nums.Length;
        int temp = 0;
        int pointer = 0;

        for(int i = 0; i<len; i++)
        {
            if(nums[i] != 0)
            {
                nums[pointer++] = nums[i];
            }
        }

        for(int i = pointer; i<len; i++)
        {
            nums[i] = 0;
        }

    }
}